internal/pkg/instrumentation/bpf/github.com/segmentio/kafka-go/producer/bpf/probe.bpf.c

```diff
@@ -201,10 +194,16 @@ int uprobe_WriteMessages(struct pt_regs *ctx) {
         // Optionally collect the topic, and always collect key
         collect_kafka_attributes(msg_ptr, &kafka_request->msgs[i], !global_topic);
         // Generate span id for each message
-        generate_random_bytes(kafka_request->msgs[i].SpanID, SPAN_ID_SIZE);
-        __builtin_memcpy(current_sc.SpanID, kafka_request->msgs[i].SpanID, SPAN_ID_SIZE);
+        if (i > 0) {
+            generate_random_bytes(kafka_request->msgs[i].sc.SpanID, SPAN_ID_SIZE);
```

Comment:
Why not generate a span ID for the first message?

Reply:
The span context (including the span ID) is currently created for the first message before this loop. Hence for each message (except the first one) we generate a new span ID and copy the trace ID and trace flags from the first one.

In the next PR, I plan to do a similar refactor for a `start_span` function which will contain similar logic to what is done in https://github.com/open-telemetry/opentelemetry-go/blob/82184aa44bb892ad95d98e09196a7c331f17797e/sdk/trace/tracer.go#L65